### PR TITLE
ButtonGroup Update

### LINKED ISSUE #295: ButtonGroup doesn't have props for onClick
We need to be able to pass in onClick as part of the `buttons` array in the ButtonGroup component so that when the items are actually clickable.

### PR DESCRIPTION
**Pull request by iheartrachie**
This PR adds onClick prop for event handling in button groups as well as fixes a border issue. Closes #295 @jmophoto 

## Before
screen shot 2016-05-05 at 12 09 02 pm

## After
screen shot 2016-05-05 at 12 08 32 pm

### CODE REVIEW

**Comment by lukeschunk** [comment]
:shipit:

File: src/components/ButtonGroup.js [hunk 1]
```diff
@@ -44,6 +45,7 @@ const ButtonGroup = React.createClass({
             <Button
               icon={button.icon}
               key={i}
+              onClick={button.onClick}
```
Comment (lukeschunk) [inline src/components/ButtonGroup.js]: sorry one thing - where is button.onClick defined?

Comment (iheartrachie) [inline src/components/ButtonGroup.js]: `button` is an object in the `buttons` array. I'm mapping through the array on line 39.

**Comment by lukeschunk** [comment]
:shipit:

**Pull request merged**